### PR TITLE
Add support for imagePullSecrets for Scylla and Agent images

### LINKED ISSUE #348: Support imagePullSecrets for Scylla image
* Feature Request

Currenly user is only allowed to use publically available Scylla images. Private builds should also be supported.

Implementation:
1. Add imagePullSecrets (scylla and agent separately) to root Spec of ScyllaCluster CRD
2. Pass it through to StatefulSet Pod Template

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:**
- Add support for imagePullSecrets for Scylla and Agent images.
- Add unit tests for StatefulSetForRack, including imagePullSecrets diffs

Resolves #348

### CODE REVIEW

File: pkg/api/scylla/v1/cluster_types.go [hunk 1]
```diff
@@ -80,6 +80,11 @@ type ScyllaClusterSpec struct {
 	// When Scylla Manager is not installed, these will be ignored.
 	// +optional
 	Backups []BackupTaskSpec `json:"backups,omitempty"`
+	// ImagePullSecrets for Scylla and Agent images.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
+	// +optional
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
```
Comment (zimnx) [inline pkg/api/scylla/v1/cluster_types.go:87]: Strategic merge patch is not supported on CRDs so patchStrategy and patchMergeKey tags aren't controlling anything. Let's keep tags clean.

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 2]
```diff
@@ -314,3 +315,66 @@ func TestMemberService(t *testing.T) {
...
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := StatefulSetForRack(*tc.rack, tc.scyllaCluster, tc.existingSts, tc.sidecarImage)
+
+			if err != nil {
+				t.Errorf("error: %s", err.Error())
+				return
+			}
+
+			if !apiequality.Semantic.DeepEqual(got.Spec.Template.Spec.ImagePullSecrets, tc.scyllaCluster.Spec.ImagePullSecrets) {
```
Comment (zimnx) [inline pkg/controller/scyllacluster/resource/resource_test.go:374]: lets compare entire sts.

**Review by zimnx — COMMENTED** [review]
Thanks, please rename PR title and commit msg to follow current naming convention. 
Please also extract autogenerated changes to separate commit.

File: pkg/api/scylla/v1/cluster_types.go (hunk 1 above)
Comment (rzetelskik) [reply pkg/api/scylla/v1/cluster_types.go:87]: done

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 2 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:374]: done

File: pkg/api/scylla/v1/cluster_types.go [hunk 3]
```diff
@@ -80,6 +80,11 @@ type ScyllaClusterSpec struct {
 	// When Scylla Manager is not installed, these will be ignored.
 	// +optional
 	Backups []BackupTaskSpec `json:"backups,omitempty"`
+	// ImagePullSecrets for Scylla and Agent images.
+	// +patchMergeKey=name
+	// +patchStrategy=merge
```
Comment (zimnx) [inline pkg/api/scylla/v1/cluster_types.go:85]: These are not needed anymore

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 4]
```diff
@@ -314,3 +320,338 @@ func TestMemberService(t *testing.T) {
...
+
+	placement := basicRack.Placement
+	if placement == nil {
+		placement = &scyllav1.PlacementSpec{}
+	}
+	opt := true
+
+	storageCapacity, _ := resource.ParseQuantity(basicRack.Storage.Capacity)
+
+	expectedSts := &appsv1.StatefulSet{
```
Comment (zimnx) [inline pkg/controller/scyllacluster/resource/resource_test.go:367]: oh this is painful to work with. Do something simillar to this approach: https://github.com/scylladb/scylla-operator/blob/9c745f7aef45c12cee0ad40c8a7989c50421198e/pkg/resourceapply/apps_test.go#L145-L150

File: pkg/api/scylla/v1/cluster_types.go (hunk 3 above)
Comment (rzetelskik) [reply pkg/api/scylla/v1/cluster_types.go]: removed

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 4 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: done

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 5]
```diff
@@ -314,3 +319,419 @@ func TestMemberService(t *testing.T) {
...
+			scyllaCluster: func() *scyllav1.ScyllaCluster {
+				sc := basicSC()
+				sc.Spec.ImagePullSecrets = basicImagePullSecrets()
+				return sc
+			}(),
+			existingSts:  existingSts,
+			sidecarImage: sidecarImage,
+			expectedSts: func() *appsv1.StatefulSet {
+				sts := newSts()
+				sts.Spec.Template.Spec.ImagePullSecrets = basicImagePullSecrets()
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:716]: would it be better to inline the expected value for the test? if someone accidentally makes the function `basicImagePullSecrets()` to return empty array, the test will still work, although it claims `"non-nil ImagePullSecrets`

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 6]
```diff
@@ -314,3 +319,419 @@ func TestMemberService(t *testing.T) {
...
+	basicImagePullSecrets := func() []corev1.LocalObjectReference {
+		imagePullSecrets := []corev1.LocalObjectReference{
+			{
+				Name: "basic-secrets",
+			},
+		}
+		return imagePullSecrets
+	}
+
+	newSts := func() *appsv1.StatefulSet {
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:376]: nit: `newExpectedStatefulSet`
would it be better to inline the values to the changes are explicit and reviewable?

File: pkg/api/scylla/v1/cluster_types.go [hunk 7]
```diff
@@ -80,6 +80,9 @@ type ScyllaClusterSpec struct {
 	// When Scylla Manager is not installed, these will be ignored.
 	// +optional
 	Backups []BackupTaskSpec `json:"backups,omitempty"`
+	// ImagePullSecrets for Scylla and Agent images.
```
Comment (tnozicka) [inline pkg/api/scylla/v1/cluster_types.go:83]: nit: `imagePullSecrets is an optional list of references to secrets in the same namespace to use for pulling Scylla and Agent images.`

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 8]
```diff
@@ -1,6 +1,11 @@
 package resource
 
 import (
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:5]: done

File: pkg/api/scylla/v1/cluster_types.go (hunk 7 above)
Comment (rzetelskik) [reply pkg/api/scylla/v1/cluster_types.go:83]: done

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 6 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: done

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 5 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: done

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 9]
```diff
@@ -314,3 +319,419 @@ func TestMemberService(t *testing.T) {
...
+				return sts
+			}(),
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			got, err := StatefulSetForRack(tc.rack, tc.scyllaCluster, tc.existingSts, tc.sidecarImage)
+
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: done

**Review by tnozicka — CHANGES_REQUESTED** [review]

**Comment by tnozicka** [comment]
API approved

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 10]
```diff
@@ -314,3 +316,91 @@ func TestMemberService(t *testing.T) {
...
 }
+
+func TestStatefulSetForRack(t *testing.T) {
+	basicRack := scyllav1.RackSpec{
+		Name: "rack",
+		Storage: scyllav1.StorageSpec{
+			Capacity: "1Gi",
+		},
+	}
+	var basicExistingStatefulSet *appsv1.StatefulSet = nil
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: changed to func newBasicExistingStatefulSet()

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 11]
```diff
@@ -314,3 +316,91 @@ func TestMemberService(t *testing.T) {
...
+				},
+			},
+			Status: scyllav1.ScyllaClusterStatus{
+				Racks: map[string]scyllav1.RackStatus{},
+			},
+		}
+	}
+
+	newExpectedStatefulSet := func() *appsv1.StatefulSet {
+		sts, _ := StatefulSetForRack(basicRack, newBasicScyllaCluster(), basicExistingStatefulSet, basicSidecarImage)
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: changed to hard-coded expected value

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 12]
```diff
@@ -314,3 +321,426 @@ func TestMemberService(t *testing.T) {
...
+func TestStatefulSetForRack(t *testing.T) {
+	basicRack := scyllav1.RackSpec{
+		Name: "rack",
+		Storage: scyllav1.StorageSpec{
+			Capacity: "1Gi",
+		},
+	}
+
+	newBasicExistingStatefulSet := func() *appsv1.StatefulSet {
+		return &appsv1.StatefulSet{}
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:334]: this doesn't deserve a function

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 13]
```diff
@@ -314,3 +321,426 @@ func TestMemberService(t *testing.T) {
...
+		}
+	}
+
+	newBasicStatefulSetLabelsWithVersion := func() map[string]string {
+		m := newBasicStatefulSetLabels()
+		m["scylla/scylla-version"] = ""
+		return m
+	}
+
+	newBasicExpectedStatefulSet := func() *appsv1.StatefulSet {
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:377]: this doesn't need the "expected" part, you'd use it in both the existing and expected parts

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 14]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+							},
+						},
+						InitContainers: []corev1.Container{
+							{
+								Name:            "sidecar-injection",
+								ImagePullPolicy: "IfNotPresent",
+								Command: []string{
+									"/bin/sh",
+									"-c",
+									fmt.Sprintf("cp -a /usr/bin/scylla-operator %s", naming.SharedDirName),
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:456]: don't use external variables like `naming.SharedDirName` in expected data, if the variable is change the change here should be explicit so the fallout is know and better catched on reviews

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 15]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+										Name:          "cql-ssl",
+										ContainerPort: 9142,
+									},
+									{
+										Name:          "thrift",
+										ContainerPort: 9160,
+									},
+								},
+								Command: []string{
+									path.Join(naming.SharedDirName, "scylla-operator"),
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:513]: same here

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 16]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+											FieldRef: &corev1.ObjectFieldSelector{
+												FieldPath: "metadata.name",
+											},
+										},
+									},
+									{
+										Name: "CPU_COUNT",
+										ValueFrom: &corev1.EnvVarSource{
+											ResourceFieldRef: &corev1.ResourceFieldSelector{
+												ContainerName: naming.ScyllaContainerName,
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:533]: inline

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 17]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+												Resource:      "limits.cpu",
+												Divisor:       resource.MustParse("1"),
+											},
+										},
+									},
+								},
+								Resources: basicRack.Resources,
+								VolumeMounts: []corev1.VolumeMount{
+									{
+										Name:      naming.PVCTemplateName,
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:543]: here and forwards

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 18]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+					Type: appsv1.RollingUpdateStatefulSetStrategyType,
+					RollingUpdate: &appsv1.RollingUpdateStatefulSetStrategy{
+						Partition: pointer.Int32Ptr(0),
+					},
+				},
+			},
+		}
+	}
+
+	newNonNilImagePullSecrets := func() []corev1.LocalObjectReference {
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:682]: nit: I'd find it more readable we inline short functions like that, but that might be opinionated. in the other case the nonNil prefix is kind of implicit, I'd not assume you make a function to create nil

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 19]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        basicSidecarImage,
+			expectedStatefulSet: nil,
+			expectedError: func() error {
+				_, err := resource.ParseQuantity("")
+				return fmt.Errorf("cannot parse %q: %v", "", err)
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:780]: nit: I'd just put the string inside, calling the function assumes to much internals that could change while the string stays constant

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 20]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
 		})
 	}
 }
+
+func TestStatefulSetForRack(t *testing.T) {
+	basicRack := scyllav1.RackSpec{
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:326]: let's not edit a global variable, assignments are not deep copies

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 21]
```diff
@@ -314,3 +321,590 @@ func TestMemberService(t *testing.T) {
...
+				sts := newBasicStatefulSet()
+				sts.Spec.Template.Annotations[naming.ForceRedeploymentReasonAnnotation] = "reason"
+				return sts
+			}(),
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:896]: let's add t.Parallel() in this case, this should force the global vars out

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 18 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:682]: I was taking example from how it's done in existing test, e.g. [cluster_validation_test](https://github.com/scylladb/scylla-operator/blob/603327d94c3ee0b58d3dc46fa8260588f003fb97/pkg/api/validation/cluster_validation_test.go#L30). Inlining would require duplicating the anonymous function in each test case. What would you change the "nonNil" to?

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 19 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:780]: compared string + added cmpopts.EquateErrors() for recommended error comparison

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 20 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:326]: changed to func newBasicRack

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 21 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:896]: added t.Parallel()

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 22]
```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
...
+			},
+		}
+	}
+
+	tt := []struct {
+		name                string
+		rack                scyllav1.RackSpec
+		scyllaCluster       *scyllav1.ScyllaCluster
+		existingStatefulSet *appsv1.StatefulSet
+		sidecarImage        string
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:750]: sidecarImage is always set to `""`, so you could just give the function a value (ideally something image like)

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 23]
```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
...
+			expectedStatefulSet: func() *appsv1.StatefulSet {
+				sts := newBasicStatefulSet()
+				sts.Spec.Template.Annotations[naming.ForceRedeploymentReasonAnnotation] = "reason"
+				return sts
+			}(),
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range tt {
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:890]: tc will move on for parallel runs, you need `tc := tc` or indexing

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 24]
```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
...
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := StatefulSetForRack(tc.rack, tc.scyllaCluster, tc.existingStatefulSet, tc.sidecarImage)
+
+			if !reflect.DeepEqual(err, tc.expectedError) {
+				t.Errorf("expected and actual errors differ: %s",
+					cmp.Diff(tc.expectedError, err, cmpopts.EquateErrors()))
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:897]: I guess with EquateErrors even suberrors would pass

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 25]
```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
...
+				r.Storage = scyllav1.StorageSpec{
+					Capacity: "",
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: &appsv1.StatefulSet{},
+			sidecarImage:        "",
+			expectedStatefulSet: nil,
+			expectedError:       fmt.Errorf(`cannot parse "": quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'`),
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:776]: given the error is public, you could use ```fmt.Errorf(`cannot parse "": %s`, ErrFormatWrong)``` or ```fmt.Errorf(`cannot parse "": %s`, errors.New("quantities must match the regular expression '^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$'"))```

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 26]
```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
...
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			got, err := StatefulSetForRack(tc.rack, tc.scyllaCluster, tc.existingStatefulSet, tc.sidecarImage)
+
+			if !reflect.DeepEqual(err, tc.expectedError) {
+				t.Errorf("expected and actual errors differ: %s",
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:896]: guess this could be fatal

**Review by tnozicka — COMMENTED** [review]
please split generated files and go.mod + vendor into separate commits

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 18 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: Thanks, I changed the shorter functions to inline and renamed the other ones

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 27]
```diff
@@ -314,3 +320,587 @@ func TestMemberService(t *testing.T) {
...
+		existingStatefulSet *appsv1.StatefulSet
+		sidecarImage        string
+		expectedStatefulSet *appsv1.StatefulSet
+		expectedError       error
+	}{
+		{
+			name:                "new StatefulSet",
+			rack:                newBasicRack(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: &appsv1.StatefulSet{},
```
[the remark this answers is not in the document]
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: changed to nil

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 22 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: fixed

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 23 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go:884]: fixed

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 24 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: you're right, I removed cmopts.EquateErrors()

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 25 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: fixed

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 26 above)
Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: fixed

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 28]
```diff
@@ -314,3 +319,583 @@ func TestMemberService(t *testing.T) {
...
+				r := newBasicRack()
+				r.Storage = scyllav1.StorageSpec{
+					Capacity: "",
+				}
+				return r
+			}(),
+			scyllaCluster:       newBasicScyllaCluster(),
+			existingStatefulSet: nil,
+			expectedStatefulSet: nil,
+			expectedError:       fmt.Errorf(`cannot parse "": %s`, resource.ErrFormatWrong),
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:754]: nit: errors print with `%v`

File: pkg/controller/scyllacluster/resource/resource_test.go [hunk 29]
```diff
@@ -314,3 +319,583 @@ func TestMemberService(t *testing.T) {
...
+				return sts
+			}(),
+			expectedError: nil,
+		},
+	}
+
+	for _, tc := range tt {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```
Comment (tnozicka) [inline pkg/controller/scyllacluster/resource/resource_test.go:887]: nit: add a new line after

**Review by tnozicka — COMMENTED** [review]
nits, lgtm otherwise

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 28 above)
Comment (tnozicka) [reply pkg/controller/scyllacluster/resource/resource_test.go:754]: or `%w`, guess that would be more fitting for this case

Comment (rzetelskik) [reply pkg/controller/scyllacluster/resource/resource_test.go]: thanks, fixed

**Review by tnozicka — APPROVED** [review]
/lgtm
thanks!

File: pkg/controller/scyllacluster/resource/resource_test.go (hunk 28 above)
Comment (tnozicka) [reply pkg/controller/scyllacluster/resource/resource_test.go]: yeah, was afraid of that, but a code change in that direction would be valid too.

**Review by zimnx — APPROVED** [review]
LGTM, thanks

**Pull request merged**